pyro/burgers/advective_fluxes.py

```diff
@@ -0,0 +1,164 @@
+
+    """
+
+    myg = my_data.grid
+
+    u = my_data.get_var("x-velocity")
+    v = my_data.get_var("y-velocity")
+
+    # cx = myg.scratch_array()
+    # cy = myg.scratch_array()
+
+    # cx.v(buf=1)[:, :] = u.v(buf=1)*dt/myg.dx
+    # cy.v(buf=1)[:, :] = v.v(buf=1)*dt/myg.dy
+
+    dtdx = dt/myg.dx
+    dtdy = dt/myg.dy
+
+    # --------------------------------------------------------------------------
+    # monotonized central differences
+    # --------------------------------------------------------------------------
+
+    limiter = rp.get_param("advection.limiter")
+
+    # Give da/dx and da/dy using input: (state, grid, direction, limiter)
+
+    ldelta_ux = reconstruction.limit(u, myg, 1, limiter)
+    ldelta_uy = reconstruction.limit(u, myg, 2, limiter)
+    ldelta_vx = reconstruction.limit(v, myg, 1, limiter)
+    ldelta_vy = reconstruction.limit(v, myg, 2, limiter)
+
+    ul_x = myg.scratch_array()
+    ur_x = myg.scratch_array()
+    ul_y = myg.scratch_array()
+    ur_y = myg.scratch_array()
+
+    vl_x = myg.scratch_array()
+    vr_x = myg.scratch_array()
+    vl_y = myg.scratch_array()
+    vr_y = myg.scratch_array()
+
+    # Determine left and right interface states in x and y.
+
+    # First compute the predictor terms for both u and v states.
+
+    ul_x.v(buf=1)[:, :] = u.ip(-1, buf=1) + 0.5*(1.0 - dtdx*u.ip(-1, buf=1))*ldelta_ux.ip(-1, buf=1)
+    ul_y.v(buf=1)[:, :] = u.jp(-1, buf=1) + 0.5*(1.0 - dtdy*v.jp(-1, buf=1))*ldelta_uy.jp(-1, buf=1)
+    ur_x.v(buf=1)[:, :] = u.v(buf=1) - 0.5*(1.0 + dtdx*u.v(buf=1))*ldelta_ux.v(buf=1)
+    ur_y.v(buf=1)[:, :] = u.v(buf=1) - 0.5*(1.0 + dtdy*v.v(buf=1))*ldelta_uy.v(buf=1)
+
+    vl_x.v(buf=1)[:, :] = v.ip(-1, buf=1) + 0.5*(1.0 - dtdx*u.ip(-1, buf=1))*ldelta_vx.ip(-1, buf=1)
+    vl_y.v(buf=1)[:, :] = v.jp(-1, buf=1) + 0.5*(1.0 - dtdy*v.jp(-1, buf=1))*ldelta_vy.jp(-1, buf=1)
+    vr_x.v(buf=1)[:, :] = v.v(buf=1) - 0.5*(1.0 + dtdx*u.v(buf=1))*ldelta_vx.v(buf=1)
+    vr_y.v(buf=1)[:, :] = v.v(buf=1) - 0.5*(1.0 + dtdy*v.v(buf=1))*ldelta_vy.v(buf=1)
+
+    # Solve Riemann's problem to get the correct transverse term
+
+    # first get the normal advective velocity through each x and y interface
+
+    uhat_adv = interface.riemann(myg, ul_x, ur_x)
+    vhat_adv = interface.riemann(myg, vl_y, vr_y)
+
+    # Upwind the l and r states using the normal advective velocity through x and y interface
+
+    ux_int = interface.upwind(myg, ul_x, ur_x, uhat_adv)
+    vx_int = interface.upwind(myg, vl_x, vr_x, uhat_adv)
+
+    uy_int = interface.upwind(myg, ul_y, ur_y, vhat_adv)
+    vy_int = interface.upwind(myg, vl_y, vr_y, vhat_adv)
+
+    # Calculate the average normal interface velocities
+
+    ubar_adv = myg.scratch_array()
+    vbar_adv = myg.scratch_array()
+
+    ubar_adv.v(buf=1)[:, :] = 0.5*(uhat_adv.v(buf=1) + uhat_adv.ip(-1, buf=1))
+    vbar_adv.v(buf=1)[:, :] = 0.5*(vhat_adv.v(buf=1) + vhat_adv.jp(-1, buf=1))
+
+    # Apply transverse correction terms:
+
```

Comment:
can you just add some comments here indicating what the tranverse term is that you are adding.  Just what is in the incompressible solver?